deployment/terraform/engine.go

```diff
@@ -89,16 +91,30 @@ func (t *Terraform) runCommand(dst io.Writer, args ...string) error {
 }
 
 func checkTerraformVersion() error {
-	if out, err := exec.Command("terraform", "version").Output(); err != nil {
+	out, err := exec.Command("terraform", "version").Output()
+	if err != nil {
 		return fmt.Errorf("could not run %q command: %w", "terraform version", err)
-	} else if re := regexp.MustCompile(`v\d+.?\d+`); !re.Match(out) {
+	}
+
+	re := regexp.MustCompile(`v\d+.?\d+`)
+	if !re.Match(out) {
 		return fmt.Errorf("could not parse terraform command output: %s", out)
-	} else if version, err := strconv.ParseFloat(string(re.FindAll(out, 1)[0][1:]), 64); err != nil {
+	}
+
+	version, err := strconv.ParseFloat(string(re.Find(out)[1:]), 64)
+	if err != nil {
 		return fmt.Errorf("could not parse terraform command output: %w", err)
-	} else if version < 0.12 {
-		return fmt.Errorf("at least terraform v0.12 is required, you have %.2f", version)
-	} else if version > 0.12 {
-		mlog.Warn(fmt.Sprintf("This tool tested with terraform v0.12, you have v%.2f. Proceed with your own risk.", version))
+	}
+	if version < minSupportedVersion {
+		return fmt.Errorf("minimum version of terraform %.2f is required, you have %.2f", minSupportedVersion, version)
+	} else if version > maxSupportedVersion {
+		mlog.Warn(fmt.Sprintf(`This tool officially supports till terraform %.2f, you have %.2f.
+Do you want to stop ? (Y/n).`, maxSupportedVersion, version))
```

Comment:
I am confused now. Why did we change the sentence? Wasn't "Do you want to proceed?" with explicit confirmation `[y/N]` a better approach?

Reply:
Because of what @isacikgoz suggested:

> Just a little touch for making confirmation more standard (I'm assuming yes should be default answer by Y/n)

Reply:
Yeah, generally default answer is `yes`. But we inverted the question now 🤣

Reply:
I'd say we can revert to proceed and keep the default to yes since it's standard approach.

Reply:
Well yeah .. we _don't_ want to proceed by default right ? I'm totally 0/5 on whatever you guys decide. Let me know :smile:

Reply:
Well, I didn't want to proceed at all and fail if the version mismatched. 
But if I take my opinion aside and think of the logic here, I guess it makes more sense to ask `Do you want to proceed?` and, since it's common practice, leave the affirmative answer as the default one.

Reply:
It aligns with `Default:open` approach, I'm fine with it.

Reply:
:+1: